hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/ImmutableSegment.java

```diff
@@ -79,7 +79,7 @@ protected ImmutableSegment(Segment segment) {
    * are going to be introduced.
    */
   protected ImmutableSegment(CellComparator comparator, MemStoreCompactorIterator iterator,
-      MemStoreLAB memStoreLAB, int numOfCells, Type type) {
+      MemStoreLAB memStoreLAB, int numOfCells, Type type, boolean doMerge) {
```

Comment:
all these doMerge flags should be part of the policy state and should not be passed throughout the code.
@sanastas this is *not* an AI for you, just a statement.